generator/internal/logging_decorator_rest_generator.cc

```diff
@@ -118,8 +119,9 @@ class $logging_rest_class_name$ : public $stub_rest_class_name$ {
 
   if (HasLongrunningMethod()) {
     // long running operation support methods
-    HeaderPrint(
-        R"""(
+    if (HasGRPCLongrunningOperation()) {
```

Comment:
I think this is too much duplication for me.

Does `$longrunning_response_type$ == "google::longrunning::Operation"` for the gRPC ones?

If not, can we just add some new substitution that is:

```cc
$longruning_get_operation_response_type$ = HasGRPCLongrunningOperation() ?
    "google::longrunning::Operation" : $longrunning_response_type$
```

Then we should only need one version of the `AsyncGetOperation` and `AsyncCancelOperation`.

---

Also we are assuming that the service only has one type of LRO. That is, it cannot return `google::longrunning::Operation` in one API and `my::bespoke::Operation` in another. Is that ok?